samples/Mvc.Client/Startup.cs

```diff
@@ -43,7 +44,13 @@ public class Startup {
             app.UseGitHubAuthentication(new GitHubAuthenticationOptions {
                 ClientId = "49e302895d8b09ea5656",
                 ClientSecret = "98f1bf028608901e9df91d64ee61536fe562064b",
-                Scope = { "user:email" }
+                Scope = {"user:email"}
+            });
+            
+            app.UseVisualStudioAuthentication(new VisualStudioAuthenticationOptions {
+                ClientId = "8DABAC41-0CAC-4BF6-BB79-670D566C55E8",
+                ClientSecret = "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Im9PdmN6NU1fN3AtSGpJS2xGWHo5M3VfVjBabyJ9.eyJjaWQiOiI4ZGFiYWM0MS0wY2FjLTRiZjYtYmI3OS02NzBkNTY2YzU1ZTgiLCJjc2kiOiI5NTRiZmQzNi05ZmUyLTRmMjYtOWFiYS00ZjM3MGUzOGI4MWIiLCJuYW1laWQiOiIxYTQ2NmYxYS0yZWViLTQzYWEtOTI5Yy0xMzQ5ZWFhMThjNWIiLCJpc3MiOiJhcHAudnNzcHMudmlzdWFsc3R1ZGlvLmNvbSIsImF1ZCI6ImFwcC52c3Nwcy52aXN1YWxzdHVkaW8uY29tIiwibmJmIjoxNDY2Mjg3NDY4LCJleHAiOjE2MjQwNTM4Njh9.x-waBWfWl5XqqMmvMwG7D4k0tkkfivOPlj0ZXGq__FJbU7SJ8oVBli1jJnrIX7B5D98hJe7s9FP8sUhIk4OPYOSCC5XqLYuVqr7Vo1-20uacEr9VN4nfDj5jJlw1JUtbK9g_pRLWzx9br3G1rxI4NpbWFpLtABz8wTEszC9Hlh9d1WdJK1GoyWLTu3VjTi2KIHlVqW_xv66nL9ImqX310iYT9ZOA4ljvLj_F3pK0Dc6EG6ZV5ryejvMkeA5Q8hypCwVPc2OiyIkn1CkwbjGGnZHEdRVLK2qpC2HP7FrcDj1-Tm8DlAIw5NUvvpxnmtZln5zuynPTaWyM74nsJ0b5wQ",
```

Comment:
How is this JWT generated? Is it something you generated yourself (which is the recommended approach when using assertions for client authentication) or is it a completely static token generated by Visual Studio Online?

The JWT has an extremely long validity period (5 years), which seems rather unusual for JWT assertions... that said, I wouldn't be surprised if it was another MSFT sauce...